packages/sv/lib/addons/_tests/mdsvex/test.ts

```diff
@@ -41,9 +41,8 @@ function addFixture(cwd: string, variant: string) {
 	}
 
 	const src = fs.readFileSync(page, 'utf8');
-	const { ast, generateCode } = parseSvelte(src);
-	const scriptAst = svelte.ensureScript(ast);
-	imports.addDefault(scriptAst, { from: './Demo.svx', as: 'Demo' });
+	const { script, generateCode } = parseSvelte(src, { ensureScript: { ext: 'ts' } });
```

Comment:
I'm not really sure about this change. After handling my other comment regarding the naming, this might make more sense, but I'm not fully convinced this is more logical than before. Also as an addon author I would wonder why there is `ensureScript` but not ensure `css`, but that's a flaw in the current implementation as well. 

I gave this a try in the [playground](https://svelte.dev/playground/542db5eb953c4590a9c703efb465aaac?version=5.46.1) to always call `ensureScript` but this is flawing the `print`ed result.

But sadly I don't see a cleaner way to this properly. So maybe this is fine as well. Just wanted to share my thoughts.

Reply:
Yes, the ~~keepNoScriptIfNoScript~~ is probably not the best idea.

I see your playground, that it's quite clear.
But in our code we need to do a bit of limbo with `ast` & `scriptAst` that is not that cool.
```ts
const ast = parse(input, { modern: true })
const scriptAst = ensureScript(ast, { langTs: true })
imports.addDefault(scriptAst, { from: './Demo.svx', as: 'Demo' });
output = print(ast).code
```

Or maybe we should just do:
```ts
const ast = parse(input, { modern: true })
ensureScript(ast, { langTs: true })
imports.addDefault(ast!, { from: './Demo.svx', as: 'Demo' });        // with ! because we ensure before
output = print(ast).code
```

Or we have the ensure in the parse... I'm 50/50 on this... Let me know what you think @manuel3108 :)

Reply:
Did you mean
```
imports.addDefault(ast.instance!, { from: './Demo.svx', as: 'Demo' }); 
```
by any chance (`ast!` vs `ast.instance!`)?

Because `ast` contains the svelte ast, which can have two js blocks (`instance` & `module`) so that wouldn't work either.

Here is an alternative for that part: https://github.com/sveltejs/cli/pull/867. It literally makes typescript narrow down the type to remove the `| null` from `ast.instance` as soon as `ensureScript` is called. See that pr's description for other thoughts. I think we should revert those changes in here.